sycl/include/sycl/ext/oneapi/experimental/ballot_group.hpp

```diff
@@ -30,7 +31,13 @@ template <typename Group>
 inline std::enable_if_t<sycl::is_group_v<std::decay_t<Group>> &&
                             std::is_same_v<Group, sycl::sub_group>,
                         ballot_group<Group>>
+#ifdef __SYCL_DEVICE_ONLY__
+get_ballot_group [[__sycl_detail__::__uses_aspects__(
+    sycl::aspect::ext_oneapi_non_uniform_groups)]] (Group group,
```

Comment:
Is this the right location to place an attribute for a function declaration?  I thought the attribute went before the return type.

Reply:
Yeah the docs suggest this, but if I put it before the return type : i.e before `std::enable_if_t..` I get:

```
error: '__uses_aspects__' attribute cannot be applied to types
   33 | [[__sycl_detail__::__uses_aspects__(
```

If I put it before the `inline` too then it works, but I can't have any variation like this that doesn't then give a warning when building the compiler like:

```
warning: extra tokens at end of #endif directive
```

I have tried putting it in a bunch of places, and the only way I can compile and run without any warnings/errors and get it working correctly is like this!

I could make it so it avoid the `#else` statement but it looks a mess, so I decided this way was best.

Reply:
According to cppreference, the attribute is allowed in either place: before the `inline` or directly after the function name.  So, I think what you have is fine.

(https://en.cppreference.com/w/cpp/language/attributes)

FWIW, I presume you could do this if you wanted to place the attribute before the `inline`:

```
template <typename Group>
#ifdef __SYCL_DEVICE_ONLY__
[[__sycl_detail__::__uses_aspects__(sycl::aspect::ext_oneapi_non_uniform_groups)]]
#endif
inline std::enable_if_t<sycl::is_group_v<std::decay_t<Group>> &&
                            std::is_same_v<Group, sycl::sub_group>,
                        ballot_group<Group>>
get_ballot_group(Group group, bool predicate);
```

Reply:
You're right this does work, I thought I tried it and it didn't. It is better like that. I'll change it.